classes/fields/pick.php

```diff
@@ -2297,7 +2297,7 @@ public function get_object_data( $object_params = null ) {
 						}
 
 						if ( 'admin_ajax_relationship' === $context ) {
-							$items[] = $this->build_dfv_field_item_data_recurse_item( $result['ID'], $result['post_title'], (object) $object_params );
+							$items[] = $this->build_dfv_field_item_data_recurse_item( $result['ID'], $result[ $search_data->field_index ], (object) $object_params );
```

Comment:
`$result['ID']` should be changed to `$result[ $search_data->field_id ]` too

Reply:
Thank you for catching this